libraries/lib-graphics/graphics/Point.h

```diff
@@ -0,0 +1,192 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  Point.h
+
+  Dmitry Vedenko
+
+**********************************************************************/
+#pragma once
+
+#include <cmath>
+#include <type_traits>
+#include <numeric>
+
+namespace graphics
+{
+
+//! A point in 2D space.
+template <typename DataType>
+struct PointType final
+{
+   DataType x {};
+   DataType y {};
+
+   PointType& operator+=(PointType rhs) noexcept
+   {
+      x += rhs.x;
+      y += rhs.y;
+
+      return *this;
+   }
+
+   PointType& operator-=(PointType rhs) noexcept
+   {
+      x -= rhs.x;
+      y -= rhs.y;
+
+      return *this;
+   }
+
+   PointType& operator*=(PointType rhs) noexcept
+   {
+      x *= rhs.x;
+      y *= rhs.y;
+
+      return *this;
+   }
+
+   PointType& operator/=(PointType rhs) noexcept
+   {
+      x /= rhs.x;
+      y /= rhs.y;
+
+      return *this;
+   }
```

Comment:
Why are the two operations above useful?  They are not usual operations you want with vectors.

Reply:
Welcome to the world of graphics :-) Probably this operation is not used though